src/javascript/app/ng-wig/ng-wig-toolbar.provider.spec.js

```diff
@@ -0,0 +1,130 @@
+            }
+        }));
+
+        it('should throw an error if command is not provided', inject(() => {
+            try {
+                toolbarProvider.addStandardButton('button1', 'My button');
+            }
+            catch(ex){
+                expect(ex).toEqual('Arguments "name", "title" and "command" are required');
+            }
+        }));
+
+        it('should not throw an error if name, title and command are provided', inject(() => {
+            expect(() => { toolbarProvider.addStandardButton('button1', 'My button', 'fakeCmd()') }).not.toThrow();
+        }));
+    });
+
+    describe('addCustomButton function', () => {
+        it('should exist', inject(() => {
+            expect(toolbarProvider.addCustomButton).toBeDefined();
+        }));
+
+        it('should throw an error if name is not provided', inject(() => {
+            try {
+                toolbarProvider.addCustomButton();
+            }
+            catch(ex){
+                expect(ex).toEqual('Arguments "name" and "pluginName" are required');
+            }
+        }));
+
+        it('should throw an error if pluginName is not provided', inject(() => {
+            try {
+                toolbarProvider.addCustomButton('button1');
+            }
+            catch(ex){
+                expect(ex).toEqual('Arguments "name" and "pluginName" are required');
+            }
+        }));
+        
+        it('should not throw an error if name and pluginName are provided', inject(() => {
+            expect(() => { toolbarProvider.addCustomButton('button1', 'my-button') }).not.toThrow();
+        }));
+    });
+
+    describe('$get function', () => {
+        it('should exist', inject(() => {
+            expect(toolbarProvider.$get).toBeDefined();
+        }));
+
+        describe('getToolbarButtons function', () => {
+            it('should exist', inject(() => {
+                expect(toolbarProvider.$get().getToolbarButtons).toBeDefined();
+            }));
+
+            it('should throw an error if provided buttons have not been added first', inject(() => {
+                try {
+                    toolbarProvider.$get().getToolbarButtons(['button1']);
+                }
+                catch(ex){
+                    expect(ex).toEqual('There is no "button1" in your library. Possible variants: list1,list2,bold,italic,link,clear-styles,forecolor,formats');
+                }
+            }));
+            
+            it('should return 8 buttons by default', inject(() => {
+                expect(toolbarProvider.$get().getToolbarButtons().length).toEqual(8);
+            }));
+
+            it('should return 1 button', inject(() => {
+                toolbarProvider.addCustomButton('button1', 'my-button');
+
+                expect(toolbarProvider.$get().getToolbarButtons(['button1']).length).toEqual(1);
+            }));
+
+            it('should add isActive function to a button', inject(() => {
```

Comment:
@stevermeister I have added a test for **isActive** function but I think the [if clause](https://github.com/stevermeister/ngWig/blob/master/src/javascript/app/ng-wig/ng-wig-toolbar.provider.js#L55) in **ngWigToolbar** provider is not needed since I don't see any way to set the function explicitly.